File: src/Application Specific/Classes/Entity Collections/MaxEngineerTime/MaxEngineerTimeSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace MaxEngineerTimes
    {
        public class MaxEngineerTimeSQL
        {
            private Sys.Database _database;

            public MaxEngineerTimeSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            public void Delete(int MaxEngineerTimeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@MaxEngineerTimeID", MaxEngineerTimeID, true);
                _database.ExecuteSP_NO_Return("MaxEngineerTime_Delete");
            }

            public MaxEngineerTime MaxEngineerTime_Get(int MaxEngineerTimeID)
            {
                _database.ClearParameter();
                _database.AddParameter("@MaxEngineerTimeID", MaxEngineerTimeID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("MaxEngineerTime_Get");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oMaxEngineerTime = new MaxEngineerTime();
                        oMaxEngineerTime.IgnoreExceptionsOnSetMethods = true;
                        oMaxEngineerTime.SetMaxEngineerTimeID = dt.Rows[0]["MaxEngineerTimeID"];
                        oMaxEngineerTime.SetEngineerID = dt.Rows[0]["EngineerID"];
                        oMaxEngineerTime.SetMondayValue = dt.Rows[0]["MondayValue"];
                        oMaxEngineerTime.SetTuesdayValue = dt.Rows[0]["TuesdayValue"];
                        oMaxEngineerTime.SetWednesdayValue = dt.Rows[0]["WednesdayValue"];
                        oMaxEngineerTime.SetThursdayValue = dt.Rows[0]["ThursdayValue"];
                        oMaxEngineerTime.SetFridayValue = dt.Rows[0]["FridayValue"];
                        oMaxEngineerTime.SetSaturdayValue = dt.Rows[0]["SaturdayValue"];
                        oMaxEngineerTime.SetSundayValue = dt.Rows[0]["SundayValue"];
                        oMaxEngineerTime.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oMaxEngineerTime.Exists = true;
                        return oMaxEngineerTime;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public DataView MaxEngineerTime_GetAll()
            {
                _database.ClearParameter();
                var dt = _database.ExecuteSP_DataTable("MaxEngineerTime_GetAll");
                dt.TableName = Sys.Enums.TableNames.tblMaxEngineerTime.ToString();
                return new DataView(dt);
            }

            public MaxEngineerTime MaxEngineerTime_GetForEngineer(int EngineerID)
            {
                _database.ClearParameter();
                _database.AddParameter("@EngineerID", EngineerID);

                // Get the datatable from the database store in dt
                var dt = _database.ExecuteSP_DataTable("MaxEngineerTime_GetForEngineer");
                if (dt is object)
                {
                    if (dt.Rows.Count > 0)
                    {
                        var oMaxEngineerTime = new MaxEngineerTime();
                        oMaxEngineerTime.IgnoreExceptionsOnSetMethods = true;
                        oMaxEngineerTime.SetMaxEngineerTimeID = dt.Rows[0]["MaxEngineerTimeID"];
                        oMaxEngineerTime.SetEngineerID = dt.Rows[0]["EngineerID"];
                        oMaxEngineerTime.SetMondayValue = dt.Rows[0]["MondayValue"];
                        oMaxEngineerTime.SetTuesdayValue = dt.Rows[0]["TuesdayValue"];
                        oMaxEngineerTime.SetWednesdayValue = dt.Rows[0]["WednesdayValue"];
                        oMaxEngineerTime.SetThursdayValue = dt.Rows[0]["ThursdayValue"];
                        oMaxEngineerTime.SetFridayValue = dt.Rows[0]["FridayValue"];
                        oMaxEngineerTime.SetSaturdayValue = dt.Rows[0]["SaturdayValue"];
                        oMaxEngineerTime.SetSundayValue = dt.Rows[0]["SundayValue"];
                        oMaxEngineerTime.SetDeleted = Conversions.ToBoolean(dt.Rows[0]["Deleted"]);
                        oMaxEngineerTime.Exists = true;
                        return oMaxEngineerTime;
                    }
                    else
                    {
                        return null;
                    }
                }
                else
                {
                    return null;
                }
            }

            public MaxEngineerTime Insert(MaxEngineerTime oMaxEngineerTime)
            {
                _database.ClearParameter();
                AddMaxEngineerTimeParametersToCommand(ref oMaxEngineerTime);
                oMaxEngineerTime.SetMaxEngineerTimeID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("MaxEngineerTime_Insert"));
                oMaxEngineerTime.Exists = true;
                return oMaxEngineerTime;
            }

            public void Update(MaxEngineerTime oMaxEngineerTime)
            {
                _database.ClearParameter();
                _database.AddParameter("@MaxEngineerTimeID", oMaxEngineerTime.MaxEngineerTimeID, true);
                AddMaxEngineerTimeParametersToCommand(ref oMaxEngineerTime);
                _database.ExecuteSP_NO_Return("MaxEngineerTime_Update");
            }

            private void AddMaxEngineerTimeParametersToCommand(ref MaxEngineerTime oMaxEngineerTime)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@EngineerID", oMaxEngineerTime.EngineerID, true);
                    withBlock.AddParameter("@MondayValue", oMaxEngineerTime.MondayValue, true);
                    withBlock.AddParameter("@TuesdayValue", oMaxEngineerTime.TuesdayValue, true);
                    withBlock.AddParameter("@WednesdayValue", oMaxEngineerTime.WednesdayValue, true);
                    withBlock.AddParameter("@ThursdayValue", oMaxEngineerTime.ThursdayValue, true);
                    withBlock.AddParameter("@FridayValue", oMaxEngineerTime.FridayValue, true);
                    withBlock.AddParameter("@SaturdayValue", oMaxEngineerTime.SaturdayValue, true);
                    withBlock.AddParameter("@SundayValue", oMaxEngineerTime.SundayValue, true);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/MaxEngineerTime/MaxEngineerTimeSQL.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace MaxEngineerTimes

Public Class MaxEngineerTimeSQL
    Private _database As Sys.Database

    Public Sub New(ByVal database As Sys.Database)
        _database = database
    End Sub

#Region "Functions"

	
    Public Sub Delete(ByVal MaxEngineerTimeID As Integer)
    		_database.ClearParameter()
            _database.AddParameter("@MaxEngineerTimeID", MaxEngineerTimeID, True)
            _database.ExecuteSP_NO_Return("MaxEngineerTime_Delete")
    End Sub

    Public Function [MaxEngineerTime_Get](ByVal MaxEngineerTimeID As Integer) As MaxEngineerTime
        _database.ClearParameter()
        _database.AddParameter("@MaxEngineerTimeID", MaxEngineerTimeID)
        
        'Get the datatable from the database store in dt
        Dim dt As DataTable = _database.ExecuteSP_DataTable("MaxEngineerTime_Get")
        
        If Not dt Is Nothing Then			
			if dt.rows.count > 0 then
				Dim oMaxEngineerTime As New MaxEngineerTime
				With oMaxEngineerTime
				    .IgnoreExceptionsOnSetMethods = True
						.SetMaxEngineerTimeID= dt.Rows(0).Item("MaxEngineerTimeID")
				.SetEngineerID= dt.Rows(0).Item("EngineerID")
				.SetMondayValue= dt.Rows(0).Item("MondayValue")
				.SetTuesdayValue= dt.Rows(0).Item("TuesdayValue")
				.SetWednesdayValue= dt.Rows(0).Item("WednesdayValue")
				.SetThursdayValue= dt.Rows(0).Item("ThursdayValue")
				.SetFridayValue= dt.Rows(0).Item("FridayValue")
				.SetSaturdayValue= dt.Rows(0).Item("SaturdayValue")
				.SetSundayValue= dt.Rows(0).Item("SundayValue")

					.SetDeleted = dt.Rows(0).Item("Deleted")               
				End With		
				oMaxEngineerTime.Exists = true				
			Return oMaxEngineerTime
			else
					Return Nothing
			end if 
			       Else
            Return Nothing
        End If
    End Function

    Public Function [MaxEngineerTime_GetAll]() As DataView
       _database.ClearParameter()
        Dim dt As DataTable = _database.ExecuteSP_DataTable("MaxEngineerTime_GetAll")
        dt.TableName = Entity.Sys.Enums.TableNames.tblMaxEngineerTime.ToString
        Return New DataView(dt)
            End Function

            Public Function MaxEngineerTime_GetForEngineer(ByVal EngineerID As Integer) As MaxEngineerTime
                _database.ClearParameter()
                _database.AddParameter("@EngineerID", EngineerID)

                'Get the datatable from the database store in dt
                Dim dt As DataTable = _database.ExecuteSP_DataTable("MaxEngineerTime_GetForEngineer")

                If Not dt Is Nothing Then
                    If dt.Rows.Count > 0 Then
                        Dim oMaxEngineerTime As New MaxEngineerTime
                        With oMaxEngineerTime
                            .IgnoreExceptionsOnSetMethods = True
                            .SetMaxEngineerTimeID = dt.Rows(0).Item("MaxEngineerTimeID")
                            .SetEngineerID = dt.Rows(0).Item("EngineerID")
                            .SetMondayValue = dt.Rows(0).Item("MondayValue")
                            .SetTuesdayValue = dt.Rows(0).Item("TuesdayValue")
                            .SetWednesdayValue = dt.Rows(0).Item("WednesdayValue")
                            .SetThursdayValue = dt.Rows(0).Item("ThursdayValue")
                            .SetFridayValue = dt.Rows(0).Item("FridayValue")
                            .SetSaturdayValue = dt.Rows(0).Item("SaturdayValue")
                            .SetSundayValue = dt.Rows(0).Item("SundayValue")

                            .SetDeleted = dt.Rows(0).Item("Deleted")
                        End With
                        oMaxEngineerTime.Exists = True
                        Return oMaxEngineerTime
                    Else
                        Return Nothing
                    End If
                Else
                    Return Nothing
                End If

            End Function
   
    Public Function [Insert](ByVal oMaxEngineerTime As MaxEngineerTime) As MaxEngineerTime
		 _database.ClearParameter()			
	     AddMaxEngineerTimeParametersToCommand(oMaxEngineerTime)	
						
		oMaxEngineerTime.SetMaxEngineerTimeID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("MaxEngineerTime_Insert"))				
        oMaxEngineerTime.Exists = True
        Return oMaxEngineerTime
    End Function   
    

    Public Sub [Update](ByVal oMaxEngineerTime As MaxEngineerTime)
		_database.ClearParameter()
		_database.AddParameter("@MaxEngineerTimeID", oMaxEngineerTime.MaxEngineerTimeID, True)  
		AddMaxEngineerTimeParametersToCommand(oMaxEngineerTime)
		_database.ExecuteSP_NO_Return("MaxEngineerTime_Update")
    End Sub



    Private Sub AddMaxEngineerTimeParametersToCommand(ByRef oMaxEngineerTime As MaxEngineerTime)    
        With _database
			.AddParameter("@EngineerID", oMaxEngineerTime.EngineerID, True)
			.AddParameter("@MondayValue", oMaxEngineerTime.MondayValue, True)
			.AddParameter("@TuesdayValue", oMaxEngineerTime.TuesdayValue, True)
			.AddParameter("@WednesdayValue", oMaxEngineerTime.WednesdayValue, True)
			.AddParameter("@ThursdayValue", oMaxEngineerTime.ThursdayValue, True)
			.AddParameter("@FridayValue", oMaxEngineerTime.FridayValue, True)
			.AddParameter("@SaturdayValue", oMaxEngineerTime.SaturdayValue, True)
			.AddParameter("@SundayValue", oMaxEngineerTime.SundayValue, True)

		
            
        End With        
    End Sub      
    

#End Region

End Class

End Namespace

End Namespace


